Modules/xSharePoint/Modules/xSharePoint.Util/xSharePoint.Util.psm1

```diff
@@ -1,19 +1,41 @@
-function Invoke-xSharePointCommand() {
+function Add-xSharePointUserToLocalAdmin() {
     [CmdletBinding()]
     param
     (
-        [parameter(Mandatory = $false)]
-        [System.Management.Automation.PSCredential]
-        $Credential,
+        [parameter(Mandatory = $true,Position=1)] [string] $UserName
+    )
 
-        [parameter(Mandatory = $false)]
-        [HashTable]
-        $Arguments,
+    $domainName = $UserName.Split('\')[0]
+    $accountName = $UserName.Split('\')[1]
```

Comment:
If you are going to do string splitting you need to be defensive with your treatment of the $userName varaible. Either you need to do a test for \ here, or ideally, validate the variable at the start of the DSC to ensure it is a match for DOMAIN\USER